### PR TITLE
lego: Implement LegoBackgroundColor

### PR DESCRIPTION
**Pull request by MishaProductions**
Also added editorconfig from matrix chat

### CODE REVIEW

**Comment by MishaProductions** [comment]
I don't really know what's causing for there to be another local variable to be added in lego1 (local_14) by the 3 atoi() calls. The RE'd function uses videomanager??
image

File: LEGO1/mxbackgroundcolor.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// OFFSET: LEGO1 0x1003bea0
+MxString *MxBackgroundColor::GetColorString()
+{
+  return &m_colorString;
+}
+
+// OFFSET: LEGO1 0x1003beb0
+void MxBackgroundColor::SetColorString(const char *colorString)
+{
+  m_colorString.operator=(colorString);
```
Comment (madebr) [inline LEGO1/mxbackgroundcolor.cpp:21]:
```suggestion
  m_colorString = colorString;
```

File: LEGO1/mxbackgroundcolor.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+// OFFSET: LEGO1 0x1003bea0
+MxString *MxBackgroundColor::GetColorString()
+{
+  return &m_colorString;
+}
+
+// OFFSET: LEGO1 0x1003beb0
+void MxBackgroundColor::SetColorString(const char *colorString)
```
Comment (madebr) [inline LEGO1/mxbackgroundcolor.cpp:19]: I doubt the names of this class.
These behave more like variables with callbacks.

The `GetName` method is used in 3 classes:
```
  100d7498, 100d74a8, 100d74b8, 100d8698
  100d86a8, 100d86b8, 100d86c8, 100da878
```
These vtables have the 1st and 3rd member in common, only the 2nd differs (which accepts a `const char*` and modifies the global state)

File: LEGO1/mxbackgroundcolor.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  virtual void SetColorString(const char *colorString);
+  virtual ~MxBackgroundColor();
+
+protected:
+  MxString m_name;
+  MxString m_colorString;
+  float h;
+  float s;
+  float v;
+};
```
Comment (madebr) [inline LEGO1/mxbackgroundcolor.h:20]: Please document which vtable exactly you think belongs to `MxBackgroundColor`.

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[*.{cpp,h}]
+indent_style = space
+indent_size = 2
```
Comment (madebr) [inline .editorconfig:3]:
```suggestion
root = true

[*.{cpp,h}]
indent_style = space
indent_size = 2
insert_final_newline = true
trim_trailing_whitespace = true
```

File: LEGO1/legobackgroundcolor.cpp [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#include "legobackgroundcolor.h"
+#include "legoomni.h"
+#include "legoutil.h"
+#include "legovideomanager.h"
+const char *Delimiter = "\t";
+const char *set = "set";
+const char *reset = "reset";
```
[the remark this answers is not in the document]
Comment (MishaProductions) [reply LEGO1/legobackgroundcolor.cpp:7]: The accuracy drops when I do that

[the remark this answers is not in the document]
Comment (madebr) [reply LEGO1/legobackgroundcolor.cpp:7]: And/or, `/GF` for read-only string pooling

[the remark this answers is not in the document]
Comment (MishaProductions) [reply LEGO1/legobackgroundcolor.cpp:7]: I tried that and it had the same result

**Comment by itsmattkc** [comment]
We've switched build systems, please merge/rebase when you get a chance.

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,6 @@
+root = true
+[*.{cpp,h}]
```
Comment (madebr) [inline .editorconfig:2]:
```suggestion
[*.{cpp,h,py,txt,editorconfig}]

```

**Comment by MishaProductions** [comment]
> We've switched build systems, please merge/rebase when you get a chance.

Done

**Comment by itsmattkc** [comment]
I merged this locally and for some reason it didn't register with this PR.

This is now merged in a66dbb8136a718e9217e00acc006ab905441c636

**Pull request closed without merging**